packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx

```diff
@@ -77,40 +70,22 @@ import { AutocompleteArrayInput } from './AutocompleteArrayInput';
  * a `setFilters` function. You can call this function to filter the results.
  */
 export const ReferenceArrayInput = (props: ReferenceArrayInputProps) => {
-    const {
-        children = defaultChildren,
-        reference,
-        sort,
-        filter = defaultFilter,
-    } = props;
+    const { children = defaultChildren, ...rest } = props;
     if (React.Children.count(children) !== 1) {
         throw new Error(
             '<ReferenceArrayInput> only accepts a single child (like <Datagrid>)'
         );
     }
 
-    const controllerProps = useReferenceArrayInputController({
-        ...props,
-        sort,
-        filter,
-    });
-
     return (
-        <ResourceContextProvider value={reference}>
-            <ChoicesContextProvider value={controllerProps}>
-                {children}
-            </ChoicesContextProvider>
-        </ResourceContextProvider>
+        <ReferenceArrayInputBase {...rest}>{children}</ReferenceArrayInputBase>
     );
 };
 
 const defaultChildren = <AutocompleteArrayInput />;
-const defaultFilter = {};
 
 export interface ReferenceArrayInputProps
-    extends InputProps,
-        UseReferenceArrayInputParams {
-    children?: ReactElement;
+    extends Omit<ReferenceArrayInputBaseProps, 'children'> {
+    children?: React.ReactNode;
```

Comment:
why not simply

```suggestion
    extends ReferenceArrayInputBaseProps {
```

?